### ISSUE TITLE
In mobile, Popover (floating) elements to appear as bottom sheets

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-design-system#3737]
**monday.com sync:** #9469736524

### Description
Update components that use a "floating" element (often using Popper ors Popover) so that the floating element appears as [a sheet](https://material.io/components/sheets-bottom) when in mobile.

- Dropdown
- Combobox
- DatePickerInput
- ColorPickerInput
- PickList and ValueList menu-actions slot
- Popover itself
- and probably others

Do NOT include Tooltip

Also maybe refactor Modal to use Popover or Popper and incorporate this behavior.

### Acceptance Criteria 
- This behavior is default
- Include an opt-out prop. Name TBD

#### Which Component
See list above.

#### Example Use Case
Any time a floating element is normally used in desktop/tablet sizes.

**Comment by avezina** [comment]
This would be a great enhancement and would be helpful for cases where the window is smaller like in the [JSAPI Utility Network Trace widget](https://developers.arcgis.com/javascript/latest/api-reference/esri-widgets-UtilityNetworkTrace.html).

Right now we are seeing issues with the Combobox dropdown component where the dropdown is hidden within the window which adds a scroll to the whole window UI. The length of the dropdown may vary based on the user's configuration and it would not be possible to set a minimum height to the window to fully resolve this.
![image](https://user-images.githubusercontent.com/50338557/149426191-d97cd962-01e4-4b76-bd44-36c71e802cb8.png)

The sheet approach would resolve this case, and would also improve the usability of widgets like these on a mobile device.

**Comment by jcfranco** [comment]
@avezina Any floating-ui-owning component will not escape scrolling containers by default. Can you try using `overlay-positioning="fixed"` on combobox for the issue you're observing?

**Comment by avezina** [comment]
@jcfranco looks like that works!! Thanks! 
![image](https://user-images.githubusercontent.com/50338557/203198760-d559f654-d24b-47a4-8eb4-89f61d93d9bc.png)

**Comment by macandcheese** [comment]
@Esri/calcite-designers @Esri/calcite-pes propose closing this as out of date. Users can use Sheet or otherwise absolutely position the Popovers with reference elements.

**Comment by geospatialem** [comment]
Agreed, closing the above out as unplanned. One example below using `sheet` in the `position="block-end"`: https://codepen.io/geospatialem/pen/WbxoMZB

```html
<calcite-sheet label="Confirmation" position="block-end" display-mode="float" id="example-sheet">
    <calcite-panel heading="Are you sure?">
        <calcite-button width="full">Yes, proceed</calcite-button>
        <calcite-button width="full" kind="neutral">
            Wait, not yet.
        </calcite-button>
    </calcite-panel>
</calcite-sheet>
```
